src/rules/prefer-repository-shorthand.ts

```diff
@@ -0,0 +1,99 @@
+import type ESTree from "estree";
+import type { AST as JsonAST } from "jsonc-eslint-parser";
+
+import { createRule } from "../createRule.js";
+
+const githubUrlRegex =
+	/^(?:git\+)?(?:ssh:\/\/git@|http?s:\/\/)?(?:www\.)?github\.com\//;
+
+const isGitHubUrl = (url: string) => githubUrlRegex.test(url);
+
+const cleanGitHubUrl = (url: string) =>
+	url.replace(githubUrlRegex, "").replace(/\.git$/, "");
+
+export default createRule({
+	create(context) {
+		return {
+			JSONProperty: (node) => {
+				if (
+					node.key.type === "JSONLiteral" &&
+					node.key.value === "repository" &&
+					node.parent.parent.parent.type === "Program"
+				) {
```

Comment:
This entire function consists of the `if` - it'd be cleaner IMO to invert its logic to reduce nesting. Filed an issue on create-typescript-app to lint for such a thing: https://github.com/JoshuaKGoldberg/create-typescript-app/issues/1267

I can do this manually myself in the meantime, no worries.